### DISCUSSION TITLE
Integration with Strawberry GraphQL

### DISCUSSION THREAD

**Discussion by mfruhner** [fief-dev/fief#114] [Q&A]
Hello,
I recently found this project and really like it so far.
I integrated it with FastAPI which worked quite well, but now I would like to shift my FastAPI Rest endpoints to GraphQL (still via FastAPI with [Strawberry](https://strawberry.rocks/docs/integrations/fastapi)). 
I managed to inject the auth.current_user dependency into a strawberry context and query user details via GraphQL.

However, the GET route /graphql, which hosts everything GraphQL related is now also protected, which denies me access to the GraphiQL Dashboard.
Is there a way to exclude specific routes/Methods from the authentication check?

I already tested the `optional` flag, but this didn't seem to work at all. I opened an issue in the python-client repo.

Thanks!

**Comment by fief-bailiff** [comment]
Hail, @mfruhner 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
Hi @mfruhner 👋 

The `optional` flag seems the way to go, in my opinion. It should work now with the fix I've made on the example (as mentioned in https://github.com/fief-dev/fief-python/issues/7): https://docs.fief.dev/integrate/python/fastapi/#__code_2_annotation_5
